purchase_discount/models/purchase_order.py

```diff
@@ -1,32 +1,80 @@
 # -*- coding: utf-8 -*-
-# © 2004-2009 Tiny SPRL (<http://tiny.be>).
-# © 2015 Pedro M. Baeza
-# © 2016 ACSONE SA/NV (<http://acsone.eu>)
+# Copyright 2004-2009 Tiny SPRL (<http://tiny.be>).
+# Copyright 2016 ACSONE SA/NV (<http://acsone.eu>)
+# Copyright 2015-2017 Tecnativa - Pedro M. Baeza
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
-from openerp import api, fields, models
-import openerp.addons.decimal_precision as dp
+
+from odoo import api, fields, models
+import odoo.addons.decimal_precision as dp
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    @api.depends('order_line.price_total')
+    def _amount_all(self):
+        orders2recalculate = self.filtered(lambda x: (
+            x.company_id.tax_calculation_rounding_method ==
+            'round_globally' and any(x.mapped('order_line.discount'))
+        ))
+        for order in orders2recalculate:
+            vals = {}
+            for line in order.order_line.filtered('discount'):
+                vals[line] = line.price_unit
+                line.price_unit *= (1 - line.discount) / 100.0
```

Comment:
@pedrobaeza If you'd took this calculation into a separte method it'd make extensibility way more simple. Something of the like:
```
    def _compute_discount(self, price):
        return price * (1 - self.discount or 0.0 / 100.0)
```